src/utility/Ex3.py

```diff
@@ -0,0 +1,230 @@
+        else:
+            return [sample.tolist(), n]
+
+def generate_events(data) -> list[list, int]: #Return eventList and size of list
+    n = data["eventDistribution"]["distribution"]["normal"]["samples"] #n = 43
+    k = data["eventDistribution"]["distribution"]["normal"]["numberOfValues"] #k = 200
+    min_value = data["eventDistribution"]["distribution"]["normal"]["minValue"] #min_value = -1
+    max_value = data["eventDistribution"]["distribution"]["normal"]["maxValue"] #max_value = 1
+    emo_type_count = data["eventDistribution"]["distribution"]["normal"]["numberOfFields"] # emo_type_count = 6
+    
+    mean = (min_value + max_value) / 2 #Mean
+    std = (max_value - min_value) / 6 #Standard deviation, both can be modified
+
+    allEvents = []
+
+    i = 0
+    while (i < emo_type_count): #6 types of emotion
+        sample = np.random.normal(mean, std, n) #Samples follow normal distribution
+        sample = np.round(sample, 1) #1 significant floating point
+        
+        stat, p_value = stats.shapiro(sample)  #Calculate p_value
+        alpha = 0.05 #Level of significant
+        if p_value < alpha:
+            continue
+        allEvents.append(sample)
+        i += 1
+
+    events = []
+
+    #Standardize impact of events and convert 6xn -> nx6 array
+    for i in range(0, n):
+        if(allEvents[0][i] < 0):
+            events.append([(-1)*abs(allEvents[j][i]) for j in range(0, emo_type_count) ])
+        else:
+            events.append([abs(allEvents[j][i]) for j in range(0, emo_type_count)])
+    return [events, n]
+
+def generate_ages(data) -> list[list, int]: #Return ageList and size of list
+    # n = data["ageDistribution"]["distribution"]["normal"]["samples"] # Using jsonref
+    n = data["numOfAgents"]["value"] #n = numOfAgents
+    k = data["ageDistribution"]["distribution"]["normal"]["numberOfValues"]
+    min_value = data["ageDistribution"]["distribution"]["normal"]["minValue"] # min_value = 5
+    max_value = data["ageDistribution"]["distribution"]["normal"]["maxValue"] # max_value = 104
+
+    mean = (min_value + max_value) / 2 # Mean
+    std = (max_value - min_value) / 6 # Standard deviation
+    while (True):
+        sample = np.random.normal(mean, std, n) #Samples follow normal distribution
+        sample = np.clip(sample, min_value, max_value) # Value trimming
+        sample = np.round(sample, 1)
+
+        stat, p_value = stats.kstest(sample, 'norm', args=(mean, std)) # Calc p_value
+
+        alpha = 0.05 # Level of significant
+        if p_value < alpha:
+            continue
+        else:
+            f = lambda x: ((x >= 23) & (x <= 61)) #Personels' age range: 23 <= x <= 61
+            cond = f(sample)
+            sample = np.concatenate((np.extract(~cond, sample), np.extract(cond, sample)))
+            #Put special value of age LAST, prioritize Personels' age in this case
+            return [sample.tolist(), n]
+
+def generate_pedestrians(data):
```

Comment:
Mình gen các pedestrian bằng cách gen các đối tượng con của lớp đó.

Reply:
That's basically how we're doing it!